src/ui/public/agg_response/tabify/tabify.js

```diff
@@ -37,13 +37,14 @@ export function AggResponseTabifyProvider(Private, Notifier) {
       case 'buckets':
         const buckets = new Buckets(bucket[agg.id]);
         if (buckets.length) {
+          const reorderedBuckets = buckets.orderBucketsAccordingToFilterParams(agg.params);
           const splitting = write.canSplit && agg.schema.name === 'split';
           if (splitting) {
-            write.split(agg, buckets, function forEachBucket(subBucket, key) {
+            write.split(agg, reorderedBuckets, function forEachBucket(subBucket, key) {
               collectBucket(write, subBucket, agg.getKey(subBucket, key), aggScale);
             });
           } else {
-            buckets.forEach(function (subBucket, key) {
+            reorderedBuckets.forEach(function (subBucket, key) {
```

Comment:
Getting `reorderedBuckets is undefined` errors on this line because `orderBucketsAccordingToFilterParams` is still mutating `this` instead of returning a value.